workflow-service/src/main/java/com/redhat/parodos/workflow/definition/service/WorkFlowDefinitionServiceImpl.java

```diff
@@ -106,30 +106,26 @@ public WorkFlowDefinitionResponseDTO save(String workFlowName, WorkFlowType work
 				.writeObjectValueAsString(convertWorkFlowParameters(workFlowParameters));
 
 		// set and save workflow definition
-		WorkFlowDefinition workFlowDefinition = workFlowDefinitionRepository.findFirstByName(workFlowName);
 		WorkFlowPropertiesDefinition propertiesDefinition = WorkFlowPropertiesDefinition.builder().build();
 		if (properties != null) {
 			propertiesDefinition.setVersion(properties.getVersion());
 		}
 
-		if (workFlowDefinition == null || !(Objects.equals(works.size(), workFlowDefinition.getNumberOfWorks())
-				&& Objects.equals(workFlowType, workFlowDefinition.getType())
-				&& Objects.equals(stringifyParameters, workFlowDefinition.getParameters())
-				&& Objects.equals(workFlowProcessingType.name(), workFlowDefinition.getProcessingType()))) {
-			// update the workflow if its properties changed
-			workFlowDefinition = workFlowDefinitionRepository
-					.save(Optional.ofNullable(workFlowDefinition).map(foundWorkflowDefinition -> {
-						foundWorkflowDefinition.setType(workFlowType);
-						foundWorkflowDefinition.setParameters(stringifyParameters);
-						foundWorkflowDefinition.setModifyDate(new Date());
-						foundWorkflowDefinition.setProperties(propertiesDefinition);
-						foundWorkflowDefinition.setProcessingType(workFlowProcessingType.name());
-						foundWorkflowDefinition.setNumberOfWorks(works.size());
-						return foundWorkflowDefinition;
-					}).orElse(WorkFlowDefinition.builder().name(workFlowName).type(workFlowType).createDate(new Date())
-							.parameters(stringifyParameters).modifyDate(new Date()).properties(propertiesDefinition)
-							.numberOfWorks(works.size()).processingType(workFlowProcessingType.name()).build()));
+		WorkFlowDefinition workFlowDefinition = workFlowDefinitionRepository.findFirstByName(workFlowName);
+		if (workFlowDefinition == null) {
+			// It's not created, so we created the minimal entity to update it
+			workFlowDefinition = WorkFlowDefinition.builder().name(workFlowName).createDate(new Date()).build();
 		}
+
+		workFlowDefinition.setType(workFlowType);
+		workFlowDefinition.setParameters(stringifyParameters);
+		workFlowDefinition.setModifyDate(new Date());
+		workFlowDefinition.setProperties(propertiesDefinition);
+		workFlowDefinition.setProcessingType(workFlowProcessingType.name());
+		workFlowDefinition.setNumberOfWorks(works.size());
+
+		workFlowDefinition = workFlowDefinitionRepository.save(workFlowDefinition);
```

Comment:
we only want to save the workflowDefinition if it's not found in db or updated.

Reply:
The thing is, does it matter in terms of complexity? At the moment, if the works change inside, but keep the same number will never be updated. 

Maybe, what we can do, it's avoid that save() and only use the last one (finally) something that I'm happy to do

Reply:
no I think this is good,   let's update it every time when app restarts